Fails to build with ffmpeg 5.0

Comskip does not build with ffmpeg 5.0:
```
gcc -DPACKAGE_NAME=\"Comskip\" -DPACKAGE_TARNAME=\"comskip\" -DPACKAGE_VERSION=\"0.82.009\" -DPACKAGE_STRING=\"Comskip\ 0.82.009\" -DPACKAGE_BUGREPORT=\"[https://github.com/erikkaashoek/Comskip/issues\](https://github.com/erikkaashoek/Comskip/issues/)" -DPACKAGE_URL=\"\" -DPACKAGE=\"comskip\" -DVERSION=\"0.82.009\" -DHAVE_LIBM=1 -DHAVE_LIBPTHREAD=1 -I.  -Wall  -I/usr/include/x86_64-linux-gnu -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL -DPROCESS_CC -DDONATOR -DHAVE_SDL -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o ccextratorwin/comskip_gui-encoding.o `test -f 'ccextratorwin/encoding.c' || echo './'`ccextratorwin/encoding.c
In file included from /usr/include/unistd.h:25,
                 from platform.h:6,
                 from mpeg2dec.c:24:
/usr/include/features.h:187:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
  187 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
      |   ^~~~~~~
mpeg2dec.c: In function ‘list_codecs’:
mpeg2dec.c:379:9: warning: implicit declaration of function ‘avcodec_register_all’ [-Wimplicit-function-declaration]
  379 |         avcodec_register_all();
      |         ^~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:380:13: warning: implicit declaration of function ‘av_codec_next’ [-Wimplicit-function-declaration]
  380 |         p = av_codec_next(NULL);
      |             ^~~~~~~~~~~~~
mpeg2dec.c:380:11: warning: assignment to ‘AVCodec *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  380 |         p = av_codec_next(NULL);
      |           ^
mpeg2dec.c:393:15: warning: assignment to ‘AVCodec *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  393 |             p = av_codec_next(p);
      |               ^
mpeg2dec.c: In function ‘retreive_frame_volume’:
mpeg2dec.c:406:66: error: ‘AVStream’ has no member named ‘codec’
  406 |     int s_per_frame = (to_pts - from_pts) * (double)(is->audio_st->codec->sample_rate+1);
      |                                                                  ^~
mpeg2dec.c:415:85: error: ‘AVStream’ has no member named ‘codec’
  415 |         buffer = & audio_buffer[(int)((from_pts - base_apts) * ((double)is->audio_st->codec->sample_rate+0.5) )];
      |                                                                                     ^~
mpeg2dec.c:428:70: error: ‘AVStream’ has no member named ‘codec’
  428 |         audio_samples -= (int)((from_pts - base_apts) * (is->audio_st->codec->sample_rate+0.5)); // incomplete frame before complete frame
      |                                                                      ^~
mpeg2dec.c:461:69: error: ‘AVStream’ has no member named ‘codec’
  461 |         top_apts = base_apts + audio_samples / (double)(is->audio_st->codec->sample_rate);
      |                                                                     ^~
mpeg2dec.c: In function ‘sound_to_frames’:
mpeg2dec.c:506:40: error: ‘AVStream’ has no member named ‘codec’
  506 |     if (old_sample_rate == is->audio_st->codec->sample_rate &&
      |                                        ^~
mpeg2dec.c:508:50: error: ‘AVStream’ has no member named ‘codec’
  508 |         || (top_apts - base_apts) * (is->audio_st->codec->sample_rate+0.5) > AUDIOBUFFER
      |                                                  ^~
mpeg2dec.c:510:65: error: ‘AVStream’ has no member named ‘codec’
  510 |         || !ISSAME(((double)audio_samples /(double)(is->audio_st->codec->sample_rate+0.5))+ base_apts, top_apts)
      |                                                                 ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:526:64: error: ‘AVStream’ has no member named ‘codec’
  526 |     if (old_sample_rate != 0 && old_sample_rate != is->audio_st->codec->sample_rate) {
      |                                                                ^~
mpeg2dec.c:527:93: error: ‘AVStream’ has no member named ‘codec’
  527 |          Debug(5, "Audio samplerate switched from %d to %d\n", old_sample_rate, is->audio_st->codec->sample_rate );
      |                                                                                             ^~
mpeg2dec.c:529:35: error: ‘AVStream’ has no member named ‘codec’
  529 |     old_sample_rate = is->audio_st->codec->sample_rate;
      |                                   ^~
mpeg2dec.c:532:90: error: ‘AVStream’ has no member named ‘codec’
  532 |     if (fabs(base_apts - (is->audio_clock - ((double)audio_samples /(double)(is->audio_st->codec->sample_rate))))> 0.0001)
      |                                                                                          ^~
mpeg2dec.c:533:85: error: ‘AVStream’ has no member named ‘codec’
  533 |         base_apts = (is->audio_clock - ((double)audio_samples /(double)(is->audio_st->codec->sample_rate)));
      |                                                                                     ^~
mpeg2dec.c:532:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  532 |     if (fabs(base_apts - (is->audio_clock - ((double)audio_samples /(double)(is->audio_st->codec->sample_rate))))> 0.0001)
      |     ^~
mpeg2dec.c:534:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  534 |         if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |         ^~
mpeg2dec.c:534:46: error: ‘AVStream’ has no member named ‘codec’
  534 |         if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                              ^~
mpeg2dec.c:559:38: error: ‘AVStream’ has no member named ‘codec’
  559 |             for (l=0;l < is->audio_st->codec->channels;l++ )
      |                                      ^~
mpeg2dec.c:567:46: error: ‘AVStream’ has no member named ‘codec’
  567 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((fb[l])++) * 64000;
      |                                              ^~
mpeg2dec.c:569:46: error: ‘AVStream’ has no member named ‘codec’
  569 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((fb[0])++) * 64000;
      |                                              ^~
mpeg2dec.c:570:60: error: ‘AVStream’ has no member named ‘codec’
  570 |                 *audio_buffer_ptr++ = volume / is->audio_st->codec->channels;
      |                                                            ^~
mpeg2dec.c:571:56: error: ‘AVStream’ has no member named ‘codec’
  571 |                 avg_volume += abs(volume / is->audio_st->codec->channels);
      |                                                        ^~
mpeg2dec.c:576:38: error: ‘AVStream’ has no member named ‘codec’
  576 |             for (l=0;l < is->audio_st->codec->channels;l++ )
      |                                      ^~
mpeg2dec.c:584:46: error: ‘AVStream’ has no member named ‘codec’
  584 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((sb[l])++);
      |                                              ^~
mpeg2dec.c:586:46: error: ‘AVStream’ has no member named ‘codec’
  586 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((sb[0])++);
      |                                              ^~
mpeg2dec.c:587:60: error: ‘AVStream’ has no member named ‘codec’
  587 |                 *audio_buffer_ptr++ = volume / is->audio_st->codec->channels;
      |                                                            ^~
mpeg2dec.c:588:56: error: ‘AVStream’ has no member named ‘codec’
  588 |                 avg_volume += abs(volume / is->audio_st->codec->channels);
      |                                                        ^~
mpeg2dec.c:594:65: error: ‘AVStream’ has no member named ‘codec’
  594 |     top_apts = base_apts + audio_samples / (double)(is->audio_st->codec->sample_rate);
      |                                                                 ^~
mpeg2dec.c: In function ‘audio_packet_process’:
mpeg2dec.c:632:44: error: ‘AVStream’ has no member named ‘codec’
  632 |     if ( !ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3
      |                                            ^~
mpeg2dec.c:645:42: error: ‘AVStream’ has no member named ‘codec’
  645 |     if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                          ^~
mpeg2dec.c:696:50: error: ‘AVStream’ has no member named ‘codec’
  696 |             if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                                  ^~
mpeg2dec.c:740:16: warning: implicit declaration of function ‘avcodec_decode_audio4’; did you mean ‘avcodec_decode_subtitle2’? [-Wimplicit-function-declaration]
  740 |         len1 = avcodec_decode_audio4(is->audio_st->codec, is->frame, &got_frame, pkt_temp);
      |                ^~~~~~~~~~~~~~~~~~~~~
      |                avcodec_decode_subtitle2
mpeg2dec.c:740:50: error: ‘AVStream’ has no member named ‘codec’
  740 |         len1 = avcodec_decode_audio4(is->audio_st->codec, is->frame, &got_frame, pkt_temp);
      |                                                  ^~
mpeg2dec.c:742:79: error: ‘AVStream’ has no member named ‘codec’
  742 |         if (prev_codec_id != -1 && (unsigned int)prev_codec_id != is->audio_st->codec->codec_id)
      |                                                                               ^~
mpeg2dec.c:746:37: error: ‘AVStream’ has no member named ‘codec’
  746 |         prev_codec_id = is->audio_st->codec->codec_id;
      |                                     ^~
mpeg2dec.c:751:29: error: ‘AVStream’ has no member named ‘codec’
  751 |             if (is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) ac3_packet_index = 0;
      |                             ^~
mpeg2dec.c:784:42: error: ‘AVStream’ has no member named ‘codec’
  784 |     if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                          ^~
mpeg2dec.c: In function ‘DoSeekRequest’:
mpeg2dec.c:1047:114: error: ‘AVFormatContext’ has no member named ‘filename’
 1047 |         fprintf(stderr, "%s error while seeking. target=%6.3f, \"%s\"\n", error_text,is->seek_pts, is->pFormatCtx->filename);
      |                                                                                                                  ^~
mpeg2dec.c:1052:123: error: ‘AVFormatContext’ has no member named ‘filename’
 1052 |             fprintf(sample_file, "%s error while seeking, target=%6.3f, \"%s\"\n", error_text,is->seek_pts, is->pFormatCtx->filename);
      |                                                                                                                           ^~
mpeg2dec.c:1067:47: error: ‘AVStream’ has no member named ‘codec’
 1067 |             avcodec_flush_buffers(is->audio_st->codec);
      |                                               ^~
mpeg2dec.c:1071:47: error: ‘AVStream’ has no member named ‘codec’
 1071 |             avcodec_flush_buffers(is->video_st->codec);
      |                                               ^~
mpeg2dec.c: In function ‘video_packet_process’:
mpeg2dec.c:1232:39: error: ‘AVStream’ has no member named ‘codec’
 1232 |     if (!hardware_decode) is->video_st->codec->flags |= AV_CODEC_FLAG_GRAY;
      |                                       ^~
mpeg2dec.c:1234:12: warning: implicit declaration of function ‘avcodec_decode_video2’; did you mean ‘avcodec_decode_subtitle2’? [-Wimplicit-function-declaration]
 1234 |     len1 = avcodec_decode_video2(is->video_st->codec, is->pFrame, &frameFinished,
      |            ^~~~~~~~~~~~~~~~~~~~~
      |            avcodec_decode_subtitle2
mpeg2dec.c:1234:46: error: ‘AVStream’ has no member named ‘codec’
 1234 |     len1 = avcodec_decode_video2(is->video_st->codec, is->pFrame, &frameFinished,
      |                                              ^~
mpeg2dec.c:1260:24: error: ‘AVStream’ has no member named ‘codec’
 1260 |         if(is->video_st->codec->framerate.den && is->video_st->codec->framerate.num)
      |                        ^~
mpeg2dec.c:1260:62: error: ‘AVStream’ has no member named ‘codec’
 1260 |         if(is->video_st->codec->framerate.den && is->video_st->codec->framerate.num)
      |                                                              ^~
mpeg2dec.c:1262:50: error: ‘AVStream’ has no member named ‘codec’
 1262 |             frame_delay = (1/ av_q2d(is->video_st->codec->framerate) ) /* * is->video_st->codec->ticks_per_frame */ ;
      |                                                  ^~
mpeg2dec.c:1266:45: error: ‘AVStream’ has no member named ‘codec’
 1266 |            frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame ;
      |                                             ^~
mpeg2dec.c:1266:79: error: ‘AVStream’ has no member named ‘codec’
 1266 |            frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame ;
      |                                                                               ^~
mpeg2dec.c:1276:17: warning: implicit declaration of function ‘av_frame_set_best_effort_timestamp’ [-Wimplicit-function-declaration]
 1276 |                 av_frame_set_best_effort_timestamp(is->pFrame, is->pFrame->pkt_pts);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:1276:76: error: ‘AVFrame’ has no member named ‘pkt_pts’; did you mean ‘pkt_dts’?
 1276 |                 av_frame_set_best_effort_timestamp(is->pFrame, is->pFrame->pkt_pts);
      |                                                                            ^~~~~~~
      |                                                                            pkt_dts
mpeg2dec.c:1277:33: warning: implicit declaration of function ‘av_frame_get_best_effort_timestamp’ [-Wimplicit-function-declaration]
 1277 |         best_effort_timestamp = av_frame_get_best_effort_timestamp(is->pFrame);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:1386:56: error: ‘AVStream’ has no member named ‘codec’
 1386 |                 if (!ISSAME(3*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay))
      |                                                        ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1387:60: error: ‘AVStream’ has no member named ‘codec’
 1387 |                     if (!ISSAME(1*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay))
      |                                                            ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1402:51: error: ‘AVStream’ has no member named ‘codec’
 1402 |             && !ISSAME(3*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
      |                                                   ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1403:51: error: ‘AVStream’ has no member named ‘codec’
 1403 |             && !ISSAME(2*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
      |                                                   ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1404:51: error: ‘AVStream’ has no member named ‘codec’
 1404 |             && !ISSAME(1*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
      |                                                   ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c: In function ‘stream_component_open’:
mpeg2dec.c:1652:49: error: ‘AVStream’ has no member named ‘codec’
 1652 |     codecCtx = pFormatCtx->streams[stream_index]->codec;
      |                                                 ^~
mpeg2dec.c:1724:15: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1724 |         codec = avcodec_find_decoder(codecCtx->codec_id);
      |               ^
mpeg2dec.c:1728:130: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1728 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_mmal");
      |                                                                                                                                  ^
mpeg2dec.c:1729:123: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1729 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_mmal");
      |                                                                                                                           ^
mpeg2dec.c:1730:125: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1730 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_mmal");
      |                                                                                                                             ^
mpeg2dec.c:1731:121: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1731 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_mmal");
      |                                                                                                                         ^
mpeg2dec.c:1735:131: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1735 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_cuvid");
      |                                                                                                                                   ^
mpeg2dec.c:1736:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1736 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_cuvid");
      |                                                                                                                            ^
mpeg2dec.c:1737:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1737 |                 if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_cuvid");
      |                                                                                                                            ^
mpeg2dec.c:1738:126: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1738 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_cuvid");
      |                                                                                                                              ^
mpeg2dec.c:1739:123: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1739 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_cuvidl") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_cuvidl");
      |                                                                                                                           ^
mpeg2dec.c:1743:131: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1743 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_vdpau");
      |                                                                                                                                   ^
mpeg2dec.c:1744:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1744 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_vdpau");
      |                                                                                                                            ^
mpeg2dec.c:1745:126: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1745 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_vdpau");
      |                                                                                                                              ^
mpeg2dec.c:1746:122: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1746 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_vdpau");
      |                                                                                                                          ^
mpeg2dec.c:1749:131: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1749 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
      |                                                                                                                                   ^
mpeg2dec.c:1750:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1750 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_dxva2");
      |                                                                                                                            ^
mpeg2dec.c:1751:126: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1751 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_dxva2");
      |                                                                                                                              ^
mpeg2dec.c:1752:122: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1752 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_dxva2");
      |                                                                                                                          ^
mpeg2dec.c:1809:32: warning: implicit declaration of function ‘av_codec_get_max_lowres’ [-Wimplicit-function-declaration]
 1809 |         codecCtx->lowres = min(av_codec_get_max_lowres(codecCtx->codec),lowres);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:1828:25: error: ‘AVStream’ has no member named ‘codec’
 1828 |             is->video_st->codec->ticks_per_frame = 1;
      |                         ^~
mpeg2dec.c: In function ‘file_open’:
mpeg2dec.c:1913:9: warning: implicit declaration of function ‘av_register_all’ [-Wimplicit-function-declaration]
 1913 |         av_register_all();
      |         ^~~~~~~~~~~~~~~
mpeg2dec.c:2013:45: error: ‘AVStream’ has no member named ‘codec’
 2013 |             is->fps = 1/(av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame );
      |                                             ^~
mpeg2dec.c:2013:79: error: ‘AVStream’ has no member named ‘codec’
 2013 |             is->fps = 1/(av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame );
      |                                                                               ^~
mpeg2dec.c: In function ‘file_close’:
mpeg2dec.c:2086:86: error: ‘AVStream’ has no member named ‘codec’
 2086 |     if (is->videoStream != -1) avcodec_close(is->pFormatCtx->streams[is->videoStream]->codec);
      |                                                                                      ^~
mpeg2dec.c:2090:86: error: ‘AVStream’ has no member named ‘codec’
 2090 |     if (is->audioStream != -1) avcodec_close(is->pFormatCtx->streams[is->audioStream]->codec);
      |                                                                                      ^~
mpeg2dec.c:2092:93: error: ‘AVStream’ has no member named ‘codec’
 2092 |     if (is->subtitleStream != -1)  avcodec_close(is->pFormatCtx->streams[is->subtitleStream]->codec);
      |                                                                                             ^~
mpeg2dec.c: In function ‘main’:
mpeg2dec.c:2387:65: error: ‘AVStream’ has no member named ‘codec’
 2387 |                         double frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame;
      |                                                                 ^~
mpeg2dec.c:2387:99: error: ‘AVStream’ has no member named ‘codec’
 2387 |                         double frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame;
      |                                                                                                   ^~
make[1]: *** [Makefile:576: comskip-mpeg2dec.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from /usr/include/unistd.h:25,
                 from platform.h:6,
                 from mpeg2dec.c:24:
/usr/include/features.h:187:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
  187 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
      |   ^~~~~~~
mpeg2dec.c: In function ‘list_codecs’:
mpeg2dec.c:379:9: warning: implicit declaration of function ‘avcodec_register_all’ [-Wimplicit-function-declaration]
  379 |         avcodec_register_all();
      |         ^~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:380:13: warning: implicit declaration of function ‘av_codec_next’ [-Wimplicit-function-declaration]
  380 |         p = av_codec_next(NULL);
      |             ^~~~~~~~~~~~~
mpeg2dec.c:380:11: warning: assignment to ‘AVCodec *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  380 |         p = av_codec_next(NULL);
      |           ^
mpeg2dec.c:393:15: warning: assignment to ‘AVCodec *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  393 |             p = av_codec_next(p);
      |               ^
mpeg2dec.c: In function ‘retreive_frame_volume’:
mpeg2dec.c:406:66: error: ‘AVStream’ has no member named ‘codec’
  406 |     int s_per_frame = (to_pts - from_pts) * (double)(is->audio_st->codec->sample_rate+1);
      |                                                                  ^~
mpeg2dec.c:415:85: error: ‘AVStream’ has no member named ‘codec’
  415 |         buffer = & audio_buffer[(int)((from_pts - base_apts) * ((double)is->audio_st->codec->sample_rate+0.5) )];
      |                                                                                     ^~
mpeg2dec.c:428:70: error: ‘AVStream’ has no member named ‘codec’
  428 |         audio_samples -= (int)((from_pts - base_apts) * (is->audio_st->codec->sample_rate+0.5)); // incomplete frame before complete frame
      |                                                                      ^~
mpeg2dec.c:461:69: error: ‘AVStream’ has no member named ‘codec’
  461 |         top_apts = base_apts + audio_samples / (double)(is->audio_st->codec->sample_rate);
      |                                                                     ^~
mpeg2dec.c: In function ‘sound_to_frames’:
mpeg2dec.c:506:40: error: ‘AVStream’ has no member named ‘codec’
  506 |     if (old_sample_rate == is->audio_st->codec->sample_rate &&
      |                                        ^~
mpeg2dec.c:508:50: error: ‘AVStream’ has no member named ‘codec’
  508 |         || (top_apts - base_apts) * (is->audio_st->codec->sample_rate+0.5) > AUDIOBUFFER
      |                                                  ^~
mpeg2dec.c:510:65: error: ‘AVStream’ has no member named ‘codec’
  510 |         || !ISSAME(((double)audio_samples /(double)(is->audio_st->codec->sample_rate+0.5))+ base_apts, top_apts)
      |                                                                 ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:526:64: error: ‘AVStream’ has no member named ‘codec’
  526 |     if (old_sample_rate != 0 && old_sample_rate != is->audio_st->codec->sample_rate) {
      |                                                                ^~
mpeg2dec.c:527:93: error: ‘AVStream’ has no member named ‘codec’
  527 |          Debug(5, "Audio samplerate switched from %d to %d\n", old_sample_rate, is->audio_st->codec->sample_rate );
      |                                                                                             ^~
mpeg2dec.c:529:35: error: ‘AVStream’ has no member named ‘codec’
  529 |     old_sample_rate = is->audio_st->codec->sample_rate;
      |                                   ^~
mpeg2dec.c:532:90: error: ‘AVStream’ has no member named ‘codec’
  532 |     if (fabs(base_apts - (is->audio_clock - ((double)audio_samples /(double)(is->audio_st->codec->sample_rate))))> 0.0001)
      |                                                                                          ^~
mpeg2dec.c:533:85: error: ‘AVStream’ has no member named ‘codec’
  533 |         base_apts = (is->audio_clock - ((double)audio_samples /(double)(is->audio_st->codec->sample_rate)));
      |                                                                                     ^~
mpeg2dec.c:532:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  532 |     if (fabs(base_apts - (is->audio_clock - ((double)audio_samples /(double)(is->audio_st->codec->sample_rate))))> 0.0001)
      |     ^~
mpeg2dec.c:534:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  534 |         if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |         ^~
mpeg2dec.c:534:46: error: ‘AVStream’ has no member named ‘codec’
  534 |         if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                              ^~
mpeg2dec.c:559:38: error: ‘AVStream’ has no member named ‘codec’
  559 |             for (l=0;l < is->audio_st->codec->channels;l++ )
      |                                      ^~
mpeg2dec.c:567:46: error: ‘AVStream’ has no member named ‘codec’
  567 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((fb[l])++) * 64000;
      |                                              ^~
mpeg2dec.c:569:46: error: ‘AVStream’ has no member named ‘codec’
  569 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((fb[0])++) * 64000;
      |                                              ^~
mpeg2dec.c:570:60: error: ‘AVStream’ has no member named ‘codec’
  570 |                 *audio_buffer_ptr++ = volume / is->audio_st->codec->channels;
      |                                                            ^~
mpeg2dec.c:571:56: error: ‘AVStream’ has no member named ‘codec’
  571 |                 avg_volume += abs(volume / is->audio_st->codec->channels);
      |                                                        ^~
mpeg2dec.c:576:38: error: ‘AVStream’ has no member named ‘codec’
  576 |             for (l=0;l < is->audio_st->codec->channels;l++ )
      |                                      ^~
mpeg2dec.c:584:46: error: ‘AVStream’ has no member named ‘codec’
  584 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((sb[l])++);
      |                                              ^~
mpeg2dec.c:586:46: error: ‘AVStream’ has no member named ‘codec’
  586 |                     for (l=0;l < is->audio_st->codec->channels;l++ ) volume += *((sb[0])++);
      |                                              ^~
mpeg2dec.c:587:60: error: ‘AVStream’ has no member named ‘codec’
  587 |                 *audio_buffer_ptr++ = volume / is->audio_st->codec->channels;
      |                                                            ^~
mpeg2dec.c:588:56: error: ‘AVStream’ has no member named ‘codec’
  588 |                 avg_volume += abs(volume / is->audio_st->codec->channels);
      |                                                        ^~
mpeg2dec.c:594:65: error: ‘AVStream’ has no member named ‘codec’
  594 |     top_apts = base_apts + audio_samples / (double)(is->audio_st->codec->sample_rate);
      |                                                                 ^~
mpeg2dec.c: In function ‘audio_packet_process’:
mpeg2dec.c:632:44: error: ‘AVStream’ has no member named ‘codec’
  632 |     if ( !ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3
      |                                            ^~
mpeg2dec.c:645:42: error: ‘AVStream’ has no member named ‘codec’
  645 |     if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                          ^~
mpeg2dec.c:696:50: error: ‘AVStream’ has no member named ‘codec’
  696 |             if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                                  ^~
mpeg2dec.c:740:16: warning: implicit declaration of function ‘avcodec_decode_audio4’; did you mean ‘avcodec_decode_subtitle2’? [-Wimplicit-function-declaration]
  740 |         len1 = avcodec_decode_audio4(is->audio_st->codec, is->frame, &got_frame, pkt_temp);
      |                ^~~~~~~~~~~~~~~~~~~~~
      |                avcodec_decode_subtitle2
mpeg2dec.c:740:50: error: ‘AVStream’ has no member named ‘codec’
  740 |         len1 = avcodec_decode_audio4(is->audio_st->codec, is->frame, &got_frame, pkt_temp);
      |                                                  ^~
mpeg2dec.c:742:79: error: ‘AVStream’ has no member named ‘codec’
  742 |         if (prev_codec_id != -1 && (unsigned int)prev_codec_id != is->audio_st->codec->codec_id)
      |                                                                               ^~
mpeg2dec.c:746:37: error: ‘AVStream’ has no member named ‘codec’
  746 |         prev_codec_id = is->audio_st->codec->codec_id;
      |                                     ^~
mpeg2dec.c:751:29: error: ‘AVStream’ has no member named ‘codec’
  751 |             if (is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) ac3_packet_index = 0;
      |                             ^~
mpeg2dec.c:784:42: error: ‘AVStream’ has no member named ‘codec’
  784 |     if (ALIGN_AC3_PACKETS && is->audio_st->codec->codec_id == AV_CODEC_ID_AC3) {
      |                                          ^~
mpeg2dec.c: In function ‘DoSeekRequest’:
mpeg2dec.c:1047:114: error: ‘AVFormatContext’ has no member named ‘filename’
 1047 |         fprintf(stderr, "%s error while seeking. target=%6.3f, \"%s\"\n", error_text,is->seek_pts, is->pFormatCtx->filename);
      |                                                                                                                  ^~
mpeg2dec.c:1052:123: error: ‘AVFormatContext’ has no member named ‘filename’
 1052 |             fprintf(sample_file, "%s error while seeking, target=%6.3f, \"%s\"\n", error_text,is->seek_pts, is->pFormatCtx->filename);
      |                                                                                                                           ^~
mpeg2dec.c:1067:47: error: ‘AVStream’ has no member named ‘codec’
 1067 |             avcodec_flush_buffers(is->audio_st->codec);
      |                                               ^~
mpeg2dec.c:1071:47: error: ‘AVStream’ has no member named ‘codec’
 1071 |             avcodec_flush_buffers(is->video_st->codec);
      |                                               ^~
mpeg2dec.c: In function ‘video_packet_process’:
mpeg2dec.c:1232:39: error: ‘AVStream’ has no member named ‘codec’
 1232 |     if (!hardware_decode) is->video_st->codec->flags |= AV_CODEC_FLAG_GRAY;
      |                                       ^~
mpeg2dec.c:1234:12: warning: implicit declaration of function ‘avcodec_decode_video2’; did you mean ‘avcodec_decode_subtitle2’? [-Wimplicit-function-declaration]
 1234 |     len1 = avcodec_decode_video2(is->video_st->codec, is->pFrame, &frameFinished,
      |            ^~~~~~~~~~~~~~~~~~~~~
      |            avcodec_decode_subtitle2
mpeg2dec.c:1234:46: error: ‘AVStream’ has no member named ‘codec’
 1234 |     len1 = avcodec_decode_video2(is->video_st->codec, is->pFrame, &frameFinished,
      |                                              ^~
mpeg2dec.c:1260:24: error: ‘AVStream’ has no member named ‘codec’
 1260 |         if(is->video_st->codec->framerate.den && is->video_st->codec->framerate.num)
      |                        ^~
mpeg2dec.c:1260:62: error: ‘AVStream’ has no member named ‘codec’
 1260 |         if(is->video_st->codec->framerate.den && is->video_st->codec->framerate.num)
      |                                                              ^~
mpeg2dec.c:1262:50: error: ‘AVStream’ has no member named ‘codec’
 1262 |             frame_delay = (1/ av_q2d(is->video_st->codec->framerate) ) /* * is->video_st->codec->ticks_per_frame */ ;
      |                                                  ^~
mpeg2dec.c:1266:45: error: ‘AVStream’ has no member named ‘codec’
 1266 |            frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame ;
      |                                             ^~
mpeg2dec.c:1266:79: error: ‘AVStream’ has no member named ‘codec’
 1266 |            frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame ;
      |                                                                               ^~
mpeg2dec.c:1276:17: warning: implicit declaration of function ‘av_frame_set_best_effort_timestamp’ [-Wimplicit-function-declaration]
 1276 |                 av_frame_set_best_effort_timestamp(is->pFrame, is->pFrame->pkt_pts);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:1276:76: error: ‘AVFrame’ has no member named ‘pkt_pts’; did you mean ‘pkt_dts’?
 1276 |                 av_frame_set_best_effort_timestamp(is->pFrame, is->pFrame->pkt_pts);
      |                                                                            ^~~~~~~
      |                                                                            pkt_dts
mpeg2dec.c:1277:33: warning: implicit declaration of function ‘av_frame_get_best_effort_timestamp’ [-Wimplicit-function-declaration]
 1277 |         best_effort_timestamp = av_frame_get_best_effort_timestamp(is->pFrame);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:1386:56: error: ‘AVStream’ has no member named ‘codec’
 1386 |                 if (!ISSAME(3*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay))
      |                                                        ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1387:60: error: ‘AVStream’ has no member named ‘codec’
 1387 |                     if (!ISSAME(1*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay))
      |                                                            ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1402:51: error: ‘AVStream’ has no member named ‘codec’
 1402 |             && !ISSAME(3*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
      |                                                   ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1403:51: error: ‘AVStream’ has no member named ‘codec’
 1403 |             && !ISSAME(2*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
      |                                                   ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c:1404:51: error: ‘AVStream’ has no member named ‘codec’
 1404 |             && !ISSAME(1*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
      |                                                   ^~
mpeg2dec.c:352:30: note: in definition of macro ‘ISSAME’
  352 | #define ISSAME(T1,T2) (fabs((T1) - (T2)) < 0.001)
      |                              ^~
mpeg2dec.c: In function ‘stream_component_open’:
mpeg2dec.c:1652:49: error: ‘AVStream’ has no member named ‘codec’
 1652 |     codecCtx = pFormatCtx->streams[stream_index]->codec;
      |                                                 ^~
mpeg2dec.c:1724:15: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1724 |         codec = avcodec_find_decoder(codecCtx->codec_id);
      |               ^
mpeg2dec.c:1728:130: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1728 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_mmal");
      |                                                                                                                                  ^
mpeg2dec.c:1729:123: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1729 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_mmal");
      |                                                                                                                           ^
mpeg2dec.c:1730:125: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1730 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_mmal");
      |                                                                                                                             ^
mpeg2dec.c:1731:121: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1731 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_mmal");
      |                                                                                                                         ^
mpeg2dec.c:1735:131: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1735 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_cuvid");
      |                                                                                                                                   ^
mpeg2dec.c:1736:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1736 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_cuvid");
      |                                                                                                                            ^
mpeg2dec.c:1737:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1737 |                 if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_cuvid");
      |                                                                                                                            ^
mpeg2dec.c:1738:126: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1738 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_cuvid");
      |                                                                                                                              ^
mpeg2dec.c:1739:123: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1739 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_cuvidl") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_cuvidl");
      |                                                                                                                           ^
mpeg2dec.c:1743:131: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1743 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_vdpau");
      |                                                                                                                                   ^
mpeg2dec.c:1744:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1744 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_vdpau");
      |                                                                                                                            ^
mpeg2dec.c:1745:126: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1745 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_vdpau");
      |                                                                                                                              ^
mpeg2dec.c:1746:122: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1746 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_vdpau");
      |                                                                                                                          ^
mpeg2dec.c:1749:131: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1749 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
      |                                                                                                                                   ^
mpeg2dec.c:1750:124: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1750 |                 if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_dxva2");
      |                                                                                                                            ^
mpeg2dec.c:1751:126: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1751 |                 if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_dxva2");
      |                                                                                                                              ^
mpeg2dec.c:1752:122: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1752 |                 if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_dxva2");
      |                                                                                                                          ^
mpeg2dec.c:1809:32: warning: implicit declaration of function ‘av_codec_get_max_lowres’ [-Wimplicit-function-declaration]
 1809 |         codecCtx->lowres = min(av_codec_get_max_lowres(codecCtx->codec),lowres);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
mpeg2dec.c:1828:25: error: ‘AVStream’ has no member named ‘codec’
 1828 |             is->video_st->codec->ticks_per_frame = 1;
      |                         ^~
mpeg2dec.c: In function ‘file_open’:
mpeg2dec.c:1913:9: warning: implicit declaration of function ‘av_register_all’ [-Wimplicit-function-declaration]
 1913 |         av_register_all();
      |         ^~~~~~~~~~~~~~~
mpeg2dec.c:2013:45: error: ‘AVStream’ has no member named ‘codec’
 2013 |             is->fps = 1/(av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame );
      |                                             ^~
mpeg2dec.c:2013:79: error: ‘AVStream’ has no member named ‘codec’
 2013 |             is->fps = 1/(av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame );
      |                                                                               ^~
mpeg2dec.c: In function ‘file_close’:
mpeg2dec.c:2086:86: error: ‘AVStream’ has no member named ‘codec’
 2086 |     if (is->videoStream != -1) avcodec_close(is->pFormatCtx->streams[is->videoStream]->codec);
      |                                                                                      ^~
mpeg2dec.c:2090:86: error: ‘AVStream’ has no member named ‘codec’
 2090 |     if (is->audioStream != -1) avcodec_close(is->pFormatCtx->streams[is->audioStream]->codec);
      |                                                                                      ^~
mpeg2dec.c:2092:93: error: ‘AVStream’ has no member named ‘codec’
 2092 |     if (is->subtitleStream != -1)  avcodec_close(is->pFormatCtx->streams[is->subtitleStream]->codec);
      |                                                                                             ^~
mpeg2dec.c: In function ‘main’:
mpeg2dec.c:2387:65: error: ‘AVStream’ has no member named ‘codec’
 2387 |                         double frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame;
      |                                                                 ^~
mpeg2dec.c:2387:99: error: ‘AVStream’ has no member named ‘codec’
 2387 |                         double frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame;
      |                                                                                                   ^~
```

--- a/configure.ac
+++ b/configure.ac
@@ -49,7 +49,7 @@ AM_CONDITIONAL([ENABLE_DEBUG], [test "x${enable_debug}" = "xyes"])
 PKG_PROG_PKG_CONFIG
 AS_IF([test "x${enable_static}" = "xyes"], PKG_CONFIG="$PKG_CONFIG --static")
 PKG_CHECK_MODULES(argtable2, [argtable2 >= 2.12])
-PKG_CHECK_MODULES(ffmpeg, [libavutil >= 54.7 libavformat >= 56.4 libavcodec >= 56.1 libswscale >= 5.0])
+PKG_CHECK_MODULES(ffmpeg, [libavutil >= 54.7 libavformat >= 57.33.100 libavcodec >= 57.37.100 libswscale >= 5.0])
 
 AS_IF([test "x${enable_gui}" = "xno"], [
     has_sdl=no

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1822,6 +1822,7 @@ int stream_component_open(VideoState *is, int stream_index)
     case AVMEDIA_TYPE_VIDEO:
         is->videoStream = stream_index;
         is->video_st = pFormatCtx->streams[stream_index];
+        is->dec_ctx = codecCtx;
 
 //          is->frame_timer = (double)av_gettime() / 1000000.0;
 //          is->frame_last_delay = 40e-3;

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1207,7 +1207,7 @@ extern char					mpegfilename[];
 int video_packet_process(VideoState *is,AVPacket *packet)
 {
     double frame_delay;
-    int len1, frameFinished;
+    int len1, frameFinished = 0;
     int repeat;
     double pts;
 //    double dts;
@@ -1236,8 +1236,7 @@ static int    prev_strange_framenum = 0;
     //is->dec_ctx.thread_type
     if (!hardware_decode) is->dec_ctx->flags |= AV_CODEC_FLAG_GRAY;
     // Decode video frame
-    len1 = avcodec_decode_video2(is->dec_ctx, is->pFrame, &frameFinished,
-                                 packet);
+    len1 = avcodec_send_packet(is->dec_ctx, packet);
 
     if (len1<0)
     {
@@ -1259,8 +1258,9 @@ static int    prev_strange_framenum = 0;
     }
 
     // Did we get a video frame?
-    if(frameFinished)
+    while ((len1 = avcodec_receive_frame(is->dec_ctx, is->pFrame)) >= 0)
     {
+        frameFinished = 1;
         // convert to 8bit
         if (is->pFrame->format == AV_PIX_FMT_YUV420P10LE) {
             is->img_convert_ctx = sws_getCachedContext(is->img_convert_ctx, is->pFrame->width, is->pFrame->height, is->pFrame->format, is->pFrame->width, is->pFrame->height, AV_PIX_FMT_YUV420P, SWS_POINT, NULL, NULL, NULL);
@@ -1578,9 +1578,9 @@ static int    prev_strange_framenum = 0;
             }
         }
 #endif
-
-        return 1;
     }
+
+    return frameFinished;
 quit:
     return 0;
 }

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -741,7 +741,7 @@ void audio_packet_process(VideoState *is, AVPacket *pkt)
  //       else
  //           avcodec_get_frame_defaults(is->frame);
 
-        len1 = avcodec_decode_audio4(is->audio_st->codec, is->frame, &got_frame, pkt_temp);
+        len1 = avcodec_decode_audio4(is->audio_ctx, is->frame, &got_frame, pkt_temp);
 
         if (prev_codec_id != -1 && (unsigned int)prev_codec_id != is->audio_st->codecpar->codec_id)
         {
@@ -1069,7 +1069,7 @@ again:
     {
         if(is->audioStream >= 0)
         {
-            avcodec_flush_buffers(is->audio_st->codec);
+            avcodec_flush_buffers(is->audio_ctx);
         }
         if(is->videoStream >= 0)
         {

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -94,7 +94,7 @@ typedef struct VideoPicture
 typedef struct VideoState
 {
     AVFormatContext *pFormatCtx;
-    AVCodecContext *dec_ctx;
+    AVCodecContext *dec_ctx, *audio_ctx, *subtitle_ctx;
     int             videoStream, audioStream, subtitleStream;
 
     int             av_sync_type;
@@ -1797,12 +1797,14 @@ int stream_component_open(VideoState *is, int stream_index)
     case AVMEDIA_TYPE_SUBTITLE:
         is->subtitleStream = stream_index;
         is->subtitle_st = pFormatCtx->streams[stream_index];
+        is->subtitle_ctx = codecCtx;
         if (demux_pid)
             selected_subtitle_pid = is->subtitle_st->id;
         break;
     case AVMEDIA_TYPE_AUDIO:
         is->audioStream = stream_index;
         is->audio_st = pFormatCtx->streams[stream_index];
+        is->audio_ctx = codecCtx;
 //          is->audio_buf_size = 0;
 //          is->audio_buf_index = 0;
 

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1073,7 +1073,7 @@ again:
         }
         if(is->videoStream >= 0)
         {
-            avcodec_flush_buffers(is->video_st->codec);
+            avcodec_flush_buffers(is->dec_ctx);
         }
     }
     is->seek_no_flush = 0;
@@ -1156,7 +1156,7 @@ nextpacket:
                     break;
                 }
 /*
-                double frame_delay = av_q2d(is->video_st->codecpar->time_base)* is->video_st->codecpar->ticks_per_frame;         // <------------------------ frame delay is the time in seconds till the next frame
+                double frame_delay = av_q2d(is->dec_ctxpar->time_base)* is->dec_ctxpar->ticks_per_frame;         // <------------------------ frame delay is the time in seconds till the next frame
                 if (is->video_clock - is->seek_pts > -frame_delay / 2.0)
                 {
                     av_packet_unref(packet);
@@ -1233,10 +1233,10 @@ static int    prev_strange_framenum = 0;
     }
     real_pts = 0.0;
     pts = 0;
-    //is->video_st->codec.thread_type
-    if (!hardware_decode) is->video_st->codec->flags |= AV_CODEC_FLAG_GRAY;
+    //is->dec_ctx.thread_type
+    if (!hardware_decode) is->dec_ctx->flags |= AV_CODEC_FLAG_GRAY;
     // Decode video frame
-    len1 = avcodec_decode_video2(is->video_st->codec, is->pFrame, &frameFinished,
+    len1 = avcodec_decode_video2(is->dec_ctx, is->pFrame, &frameFinished,
                                  packet);
 
     if (len1<0)
@@ -1275,16 +1275,16 @@ static int    prev_strange_framenum = 0;
             is->pFrame = newframe;
         }
 
-        if(is->video_st->codec->framerate.den && is->video_st->codec->framerate.num)
+        if(is->dec_ctx->framerate.den && is->dec_ctx->framerate.num)
         {
-            frame_delay = (1/ av_q2d(is->video_st->codec->framerate) ) /* * is->video_st->codec->ticks_per_frame */ ;
+            frame_delay = (1/ av_q2d(is->dec_ctx->framerate) ) /* * is->dec_ctx->ticks_per_frame */ ;
         }
         else
         {
-           frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame ;
+           frame_delay = av_q2d(is->dec_ctx->time_base) * is->dec_ctx->ticks_per_frame ;
         }
 
-//        frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame ;
+//        frame_delay = av_q2d(is->dec_ctx->time_base) * is->dec_ctx->ticks_per_frame ;
         repeat = av_stream_get_parser(is->video_st) ? av_stream_get_parser(is->video_st)->repeat_pict: 4;
 
  //       if (prev_frame_delay != 0.0 && frame_delay != prev_frame_delay)
@@ -1392,24 +1392,24 @@ static int    prev_strange_framenum = 0;
 //#define SHOW_VIDEO_TIMING
 #ifdef SHOW_VIDEO_TIMING
         if (framenum==0)
-            Debug(1,"Video timing ---------------------------------------------------\n", frame_delay/is->video_st->codec->ticks_per_frame, is->video_st->codec->ticks_per_frame, repeat, real_pts,calculated_delay);
+            Debug(1,"Video timing ---------------------------------------------------\n", frame_delay/is->dec_ctx->ticks_per_frame, is->dec_ctx->ticks_per_frame, repeat, real_pts,calculated_delay);
         else if (framenum<20)
-            Debug(1,"Video timing fr=%6.5f, tick=%d, repeat=%d, pts=%6.3f, step=%6.5f\n", frame_delay/is->video_st->codec->ticks_per_frame, is->video_st->codec->ticks_per_frame, repeat, real_pts,calculated_delay);
+            Debug(1,"Video timing fr=%6.5f, tick=%d, repeat=%d, pts=%6.3f, step=%6.5f\n", frame_delay/is->dec_ctx->ticks_per_frame, is->dec_ctx->ticks_per_frame, repeat, real_pts,calculated_delay);
 #endif // SHOW_VIDEO_TIMING
 
 
         pts_offset *= 0.9;
         if (!reviewing && timeline_repair) {
             if (framenum > 1 && fabs(calculated_delay - pts_offset - frame_delay) < 1.0) { // Allow max 0.5 second timeline jitter to be compensated
-                if (!ISSAME(3*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay))
-                    if (!ISSAME(1*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay))
+                if (!ISSAME(3*frame_delay/ is->dec_ctx->ticks_per_frame, calculated_delay))
+                    if (!ISSAME(1*frame_delay/ is->dec_ctx->ticks_per_frame, calculated_delay))
                         pts_offset = pts_offset + frame_delay - calculated_delay;
             }
         }
         else
             do_audio_repair = 0;
 
-//		Debug(0 ,"pst[%3d] = %12.3f, inter = %d, ticks = %d\n", framenum, pts/frame_delay, is->pFrame->interlaced_frame, is->video_st->codecpar->ticks_per_frame);
+//		Debug(0 ,"pst[%3d] = %12.3f, inter = %d, ticks = %d\n", framenum, pts/frame_delay, is->pFrame->interlaced_frame, is->dec_ctxpar->ticks_per_frame);
 
         pts = real_pts + pts_offset;
 
@@ -1417,9 +1417,9 @@ static int    prev_strange_framenum = 0;
 
         if (!reviewing
             && framenum > 1 && fabs(calculated_delay - frame_delay) > 0.01
-            && !ISSAME(3*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
-            && !ISSAME(2*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
-            && !ISSAME(1*frame_delay/ is->video_st->codec->ticks_per_frame, calculated_delay)
+            && !ISSAME(3*frame_delay/ is->dec_ctx->ticks_per_frame, calculated_delay)
+            && !ISSAME(2*frame_delay/ is->dec_ctx->ticks_per_frame, calculated_delay)
+            && !ISSAME(1*frame_delay/ is->dec_ctx->ticks_per_frame, calculated_delay)
             ){
             if ( (prev_strange_framenum + 1 != framenum) &&( prev_strange_step < fabs(calculated_delay - frame_delay))) {
                 Debug(8 ,"Strange video pts step of %6.5f instead of %6.5f at frame %d\n", calculated_delay+0.0000005, frame_delay+0.0000005, framenum); // Unknown strange step
@@ -1847,7 +1847,7 @@ int stream_component_open(VideoState *is, int stream_index)
 #endif
         }
         if (codecCtx->codec_id == AV_CODEC_ID_MPEG1VIDEO)
-            is->video_st->codec->ticks_per_frame = 1;
+            is->dec_ctx->ticks_per_frame = 1;
         if (demux_pid)
             selected_video_pid = is->video_st->id;
         /*
@@ -2029,7 +2029,7 @@ again:
         else
         {
             Debug(10, "Warning, no stream frame rate, deriving from codec\n");
-            is->fps = 1/(av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame );
+            is->fps = 1/(av_q2d(is->dec_ctx->time_base) * is->dec_ctx->ticks_per_frame );
         }
         set_fps( 1.0 / is->fps);
 //        Debug(1, "Stream frame rate is %5.3f f/s\n", is->fps);
@@ -2403,7 +2403,7 @@ nextpacket:
 
                     if ((live_tv && retries < live_tv_retries) /* || (selftest == 3 && retries == 0) */)
                     {
-                        double frame_delay = av_q2d(is->video_st->codec->time_base) * is->video_st->codec->ticks_per_frame;
+                        double frame_delay = av_q2d(is->dec_ctx->time_base) * is->dec_ctx->ticks_per_frame;
 //                    uint64_t retry_target;
                         if (retries == 0)
                         {

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1670,6 +1670,46 @@ int stream_component_open(VideoState *is, int stream_index)
     codecCtx = pFormatCtx->streams[stream_index]->codec;
     avcodec_close(codecCtx);
 
+	codec = avcodec_find_decoder(codecCtx->codec_id);
+
+	// If decoding in hardware try if running on a Raspberry Pi and then use it's decoder instead.
+    if (hardware_decode) {
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_mmal");
+		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_mmal");
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_mmal");
+		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_mmal");
+    }
+
+    if (use_cuvid) {
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_cuvid");
+		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_cuvid");
+		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_cuvid");
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_cuvid");
+		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_cuvidl") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_cuvidl");
+    }
+
+    if (use_vdpau) {
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_vdpau");
+		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_vdpau");
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_vdpau");
+		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_vdpau");
+		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vhevc_vdpau");
+    }
+    if (use_dxva2) {
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO &&
+            avcodec_find_decoder_by_name("mpeg2_dxva2") != NULL)
+                codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
+		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_dxva2");
+		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_dxva2");
+		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_dxva2");
+		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_dxva2");
+    }
+
+    if (codec_hw != NULL && codec_hw != codec) {
+        fprintf(stderr, "Using Codec: %s instead of %s\n", codec_hw->name, codec->name);
+        codec = codec_hw;
+    }
+
     if (codecCtx->codec_type == AVMEDIA_TYPE_VIDEO)
     {
         if (!hardware_decode) codecCtx->flags |= AV_CODEC_FLAG_GRAY;
@@ -1738,53 +1778,12 @@ int stream_component_open(VideoState *is, int stream_index)
         }
     }
 
-
-	codec = avcodec_find_decoder(codecCtx->codec_id);
-
-	// If decoding in hardware try if running on a Raspberry Pi and then use it's decoder instead.
-    if (hardware_decode) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_mmal");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_mmal");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_mmal");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_mmal");
-    }
-
-    if (use_cuvid) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_cuvidl") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_cuvidl");
-    }
-
-    if (use_vdpau) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vhevc_vdpau");
-    }
-    if (use_dxva2) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO &&
-            avcodec_find_decoder_by_name("mpeg2_dxva2") != NULL)
-                codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_dxva2");
-    }
-
     if (!hardware_decode) av_dict_set_int(&myoptions, "gray", 1, 0);
 
 
  //       av_dict_set_int(&myoptions, "fastint", 1, 0);
  //       av_dict_set_int(&myoptions, "skip_alpha", 1, 0);
 //        av_dict_set(&myoptions, "threads", "auto", 0);
-
-    if (codec_hw != NULL && codec_hw != codec) {
-        fprintf(stderr, "Using Codec: %s instead of %s\n", codec_hw->name, codec->name);
-        codec = codec_hw;
-    }
 
     if(!codec || (avcodec_open2(codecCtx, codec, &myoptions) < 0))
     {

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1649,8 +1649,8 @@ static int get_buffer(AVCodecContext *s, AVFrame *frame, int flags)
 
 int stream_component_open(VideoState *is, int stream_index)
 {
-
     AVFormatContext *pFormatCtx = is->pFormatCtx;
+    AVCodecParameters *codecPar = NULL;
     AVCodecContext *codecCtx;
     const AVCodec *codec;
     const AVCodec *codec_hw = NULL;
@@ -1668,42 +1668,43 @@ int stream_component_open(VideoState *is, int stream_index)
     // Get a pointer to the codec context for the video stream
 
     codecCtx = pFormatCtx->streams[stream_index]->codec;
+    codecPar = pFormatCtx->streams[stream_index]->codecpar;
     avcodec_close(codecCtx);
 
 	codec = avcodec_find_decoder(codecCtx->codec_id);
 
+    if (use_dxva2 && !codec_hw) {
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG2VIDEO) codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
+		if (codecPar->codec_id == AV_CODEC_ID_H264) codec_hw = avcodec_find_decoder_by_name("h264_dxva2");
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG4) codec_hw = avcodec_find_decoder_by_name("mpeg4_dxva2");
+		if (codecPar->codec_id == AV_CODEC_ID_VC1) codec_hw = avcodec_find_decoder_by_name("vc1_dxva2");
+		if (codecPar->codec_id == AV_CODEC_ID_HEVC) codec_hw = avcodec_find_decoder_by_name("hevc_dxva2");
+    }
+
+    if (use_vdpau && !codec_hw) {
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG2VIDEO) codec_hw = avcodec_find_decoder_by_name("mpeg2_vdpau");
+		if (codecPar->codec_id == AV_CODEC_ID_H264) codec_hw = avcodec_find_decoder_by_name("h264_vdpau");
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG4) codec_hw = avcodec_find_decoder_by_name("mpeg4_vdpau");
+		if (codecPar->codec_id == AV_CODEC_ID_VC1) codec_hw = avcodec_find_decoder_by_name("vc1_vdpau");
+		if (codecPar->codec_id == AV_CODEC_ID_HEVC) codec_hw = avcodec_find_decoder_by_name("hevc_vdpau");
+    }
+
+    if (use_cuvid && !codec_hw) {
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG2VIDEO) codec_hw = avcodec_find_decoder_by_name("mpeg2_cuvid");
+		if (codecPar->codec_id == AV_CODEC_ID_H264) codec_hw = avcodec_find_decoder_by_name("h264_cuvid");
+		if (codecPar->codec_id == AV_CODEC_ID_HEVC) codec_hw = avcodec_find_decoder_by_name("hevc_cuvid");
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG4) codec_hw = avcodec_find_decoder_by_name("mpeg4_cuvid");
+		if (codecPar->codec_id == AV_CODEC_ID_VC1) codec_hw = avcodec_find_decoder_by_name("vc1_cuvidl");
+    }
+
 	// If decoding in hardware try if running on a Raspberry Pi and then use it's decoder instead.
-    if (hardware_decode) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_mmal");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_mmal");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_mmal");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_mmal") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_mmal");
+    if (hardware_decode && !codec_hw) {
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG2VIDEO) codec_hw = avcodec_find_decoder_by_name("mpeg2_mmal");
+		if (codecPar->codec_id == AV_CODEC_ID_H264) codec_hw = avcodec_find_decoder_by_name("h264_mmal");
+		if (codecPar->codec_id == AV_CODEC_ID_MPEG4) codec_hw = avcodec_find_decoder_by_name("mpeg4_mmal");
+		if (codecPar->codec_id == AV_CODEC_ID_VC1) codec_hw = avcodec_find_decoder_by_name("vc1_mmal");
     }
 
-    if (use_cuvid) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_cuvid") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_cuvid");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_cuvidl") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_cuvidl");
-    }
-
-    if (use_vdpau) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO && avcodec_find_decoder_by_name("mpeg2_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg2_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_vdpau");
-		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_vdpau") != NULL) codec_hw = avcodec_find_decoder_by_name("vhevc_vdpau");
-    }
-    if (use_dxva2) {
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG2VIDEO &&
-            avcodec_find_decoder_by_name("mpeg2_dxva2") != NULL)
-                codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_H264 && avcodec_find_decoder_by_name("h264_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("h264_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_MPEG4 && avcodec_find_decoder_by_name("mpeg4_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("mpeg4_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_VC1 && avcodec_find_decoder_by_name("vc1_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("vc1_dxva2");
-		if (codecCtx->codec_id == AV_CODEC_ID_HEVC && avcodec_find_decoder_by_name("hevc_dxva2") != NULL) codec_hw = avcodec_find_decoder_by_name("hevc_dxva2");
-    }
 
     if (codec_hw != NULL && codec_hw != codec) {
         fprintf(stderr, "Using Codec: %s instead of %s\n", codec_hw->name, codec->name);

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -2105,13 +2105,13 @@ void file_close()
 //    av_freep(&ist->hwaccel_device);
 
 
-    if (is->videoStream != -1) avcodec_close(is->pFormatCtx->streams[is->videoStream]->codec);
+    if (is->dec_ctx) avcodec_close(is->dec_ctx);
     is->videoStream = -1;
 //    avcodec_free_context(&is->pFormatCtx->streams[is->videoStream]->codec);
 
-    if (is->audioStream != -1) avcodec_close(is->pFormatCtx->streams[is->audioStream]->codec);
+    if (is->audio_ctx) avcodec_close(is->audio_ctx);
     is->audioStream = -1;
-    if (is->subtitleStream != -1)  avcodec_close(is->pFormatCtx->streams[is->subtitleStream]->codec);
+    if (is->subtitle_ctx)  avcodec_close(is->subtitle_ctx);
     is->subtitleStream = -1;
 //    is->pFormatCtx = NULL;
 

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -728,20 +728,13 @@ void audio_packet_process(VideoState *is, AVPacket *pkt)
 
     initial_apts_set = 1;
 
+    len1 = avcodec_send_packet(is->audio_ctx, pkt_temp);
+
     //		fprintf(stderr, "sac = %f\n", is->audio_clock);
-    while(pkt_temp->size > 0)
+    while ((len1 = avcodec_receive_frame(is->audio_ctx, is->frame)) != AVERROR(EAGAIN))
     {
  //       data_size = STORAGE_SIZE;
-
-//        if (!is->frame)
-//        {
-            if (!(is->frame = av_frame_alloc()))
-                return;
- //       }
- //       else
- //           avcodec_get_frame_defaults(is->frame);
-
-        len1 = avcodec_decode_audio4(is->audio_ctx, is->frame, &got_frame, pkt_temp);
+        got_frame = len1 >= 0;
 
         if (prev_codec_id != -1 && (unsigned int)prev_codec_id != is->audio_st->codecpar->codec_id)
         {
@@ -782,7 +775,7 @@ void audio_packet_process(VideoState *is, AVPacket *pkt)
         }
         is->audio_clock += (double)data_size /
                            (is->frame->channels * is->frame->sample_rate * av_get_bytes_per_sample(is->frame->format));
-        av_frame_free(&(is->frame));
+        av_frame_unref(is->frame);
     }
 
     if (ALIGN_AC3_PACKETS && is->audio_st->codecpar->codec_id == AV_CODEC_ID_AC3) {
@@ -1999,6 +1992,11 @@ again:
         }
         // Dump information about file onto standard error
         if (retries == 0) av_dump_format(is->pFormatCtx, 0, is->filename, 0);
+    }
+
+    if (!is->frame) {
+        if (!(is->frame = av_frame_alloc()))
+            exit(-1);
     }
 
     if ( is->videoStream == -1)

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1652,8 +1652,8 @@ int stream_component_open(VideoState *is, int stream_index)
 
     AVFormatContext *pFormatCtx = is->pFormatCtx;
     AVCodecContext *codecCtx;
-    AVCodec *codec;
-    AVCodec *codec_hw = NULL;
+    const AVCodec *codec;
+    const AVCodec *codec_hw = NULL;
 
 
 

--- a/mpeg2dec.c
+++ b/mpeg2dec.c
@@ -1667,11 +1667,9 @@ int stream_component_open(VideoState *is, int stream_index)
 
     // Get a pointer to the codec context for the video stream
 
-    codecCtx = pFormatCtx->streams[stream_index]->codec;
     codecPar = pFormatCtx->streams[stream_index]->codecpar;
-    avcodec_close(codecCtx);
 
-	codec = avcodec_find_decoder(codecCtx->codec_id);
+	codec = avcodec_find_decoder(codecPar->codec_id);
 
     if (use_dxva2 && !codec_hw) {
 		if (codecPar->codec_id == AV_CODEC_ID_MPEG2VIDEO) codec_hw = avcodec_find_decoder_by_name("mpeg2_dxva2");
@@ -1710,6 +1708,9 @@ int stream_component_open(VideoState *is, int stream_index)
         fprintf(stderr, "Using Codec: %s instead of %s\n", codec_hw->name, codec->name);
         codec = codec_hw;
     }
+
+    codecCtx = avcodec_alloc_context3(codec);
+    avcodec_parameters_to_context(codecCtx, codecPar);
 
     if (codecCtx->codec_type == AVMEDIA_TYPE_VIDEO)
     {
@@ -2105,13 +2106,13 @@ void file_close()
 //    av_freep(&ist->hwaccel_device);
 
 
-    if (is->dec_ctx) avcodec_close(is->dec_ctx);
+    if (is->dec_ctx) avcodec_free_context(&is->dec_ctx);
     is->videoStream = -1;
 //    avcodec_free_context(&is->pFormatCtx->streams[is->videoStream]->codec);
 
-    if (is->audio_ctx) avcodec_close(is->audio_ctx);
+    if (is->audio_ctx) avcodec_free_context(&is->audio_ctx);
     is->audioStream = -1;
-    if (is->subtitle_ctx)  avcodec_close(is->subtitle_ctx);
+    if (is->subtitle_ctx)  avcodec_free_context(&is->subtitle_ctx);
     is->subtitleStream = -1;
 //    is->pFormatCtx = NULL;
 

